File: 2026POTD/JANUARY/11-01-2026/85.kt
class Solution {
    fun maximalRectangle(matrix: Array<CharArray>): Int {
        if (matrix.isEmpty()) return 0

        val cols = matrix[0].size
        val heights = IntArray(cols)
        var ans = 0

        for (row in matrix) {
            for (j in 0 until cols) {
                heights[j] = if (row[j] == '1') heights[j] + 1 else 0
            }

            val stack = java.util.Stack<Int>()
            for (i in 0..cols) {
                val h = if (i == cols) 0 else heights[i]
                while (stack.isNotEmpty() && h < heights[stack.peek()]) {
                    val height = heights[stack.pop()]
                    val width = if (stack.isEmpty()) i else i - stack.peek() - 1
                    ans = maxOf(ans, height * width)
                }
                stack.push(i)
            }
        }

        return ans
    }
}

File: 2026POTD/JANUARY/11-01-2026/85.rs
impl Solution {
    pub fn maximal_rectangle(matrix: Vec<Vec<char>>) -> i32 {
        if matrix.is_empty() {
            return 0;
        }

        let cols = matrix[0].len();
        let mut heights = vec![0; cols];
        let mut ans = 0;

        for row in matrix {
            for j in 0..cols {
                heights[j] = if row[j] == '1' { heights[j] + 1 } else { 0 };
            }

            let mut stack: Vec<usize> = Vec::new();
            for i in 0..=cols {
                let h = if i == cols { 0 } else { heights[i] };
                while let Some(&top) = stack.last() {
                    if h >= heights[top] {
                        break;
                    }
                    let height = heights[top];
                    stack.pop();
                    let width = if let Some(&prev) = stack.last() {
                        i - prev - 1
                    } else {
                        i
                    };
                    ans = ans.max((height * width) as i32);
                }
                stack.push(i);
            }
        }

        ans
    }
}